lib/floe/workflow/intrinsic_function.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module Floe
+  class Workflow
+    class IntrinsicFunction
+      INTRINSIC_FUNCTION_REGEX = /^(?<module>\w+)\.(?<function>\w+)\((?<args>.*)\)$/.freeze
+
+      class << self
+        def detect_class(payload)
+          function_module_name, function_name =
+            payload.match(INTRINSIC_FUNCTION_REGEX)
+                   .named_captures
+                   .values_at("module", "function")
+
+          begin
+            function_module = Floe::Workflow::IntrinsicFunctions.const_get(function_module_name)
+            function_module.const_get(function_name)
```

Comment:
I have one concern here.  I was looking through the [list of Intrinsic Functions](https://states-language.net/#appendix-b) and there is one called `States.Array`.  This would imply we have a `Floe::Workflow::IntrinsicFunctions::States::Array` class, and that `Array` shadowing kind of scares me (regardless of const_get or literal).  I'm wondering if instead of a full namespacing, we instead just smash the module/function together so you'd get something more like: `Floe::Workflow::IntrinsicFunctions::StatesArray`.  Thoughts?

Reply:
Hm we could have `States` be a class with methods `def StringToJson` and `def Array` rather than having classes for each intrinsic function.  I like having these separated because if/when we have our own `ManageIQ` intrinsic functions they will be nicely separated.

Reply:
maybe?  methods named with capitals will surely get confusing.  If they are class methods then, technicall `::` will call those too, which works, but would still be weird.  Reminds me of the Vmdb::BUILD method we have.

Reply:
I like going the `send` route, aka: `def StringToJson` (or `def intrinsicStringToJson` if we need to start with a lowercase). Alternatively a `case` statement may not be the most ruby thing, but that tends to be the defacto implementation for this type of construct across all programming languages (both target language and complier/interpreter language).

All intrinsics are in the same namespace: `States.*`. I think hardcoding this rather than a generic "look for a period" works best, and we can hardcode any additional namespaces in the future. Lets also avoid regular expressions and use `indexOf(")")` kind of stuff. It is way too hard to deal with properly nesting parens and regular expressions. Though this implementation may be ok for a first pass.